src/custom/utils/price.ts

```diff
@@ -296,28 +297,101 @@ export function getValidParams(params: PriceQuoteParams) {
   return { ...params, baseToken, quoteToken }
 }
 
-export function calculateFallbackPriceImpact(initialValue: string, finalValue: string) {
-  const initialValueBn = new BigNumberJs(initialValue)
-  const finalValueBn = new BigNumberJs(finalValue)
-  // TODO: use correct formula
-  // ((IV - FV) / IV / 2) * 100
-  const [numerator, denominator] = initialValueBn.minus(finalValueBn).div(initialValueBn).div('2').toFraction()
-
-  console.debug(
-    '[calculateFallbackPriceImpact]::',
-    initialValueBn.toString(10),
-    finalValueBn.toString(10),
-    numerator.toString(10),
-    denominator.toString(10)
-  )
+function _isWithinRange(numerator: BigNumberJs, denominator: BigNumberJs) {
+  // e.g 5 > 2 = true ==> within [0,1] range
+  // e.g 2 > 5 = false
+  return denominator.isGreaterThan(numerator)
+}
+
+// *** SELL ***
+// ------------------
+// Pab = abOut/abIn
+// Pba = baOut/baIn[abOut]
+// S = 1 - sqrt( Pab*Pba )
+// OR
+// S = 1 + sqrt( Pab*Pba )
+// -------------------------------------------
+// EXAMPLE
+// -------------------------------------------
+// 100A > 80B // 80B >> 56A
+// --> 80/100 = 0.8
+// --> 56/80 = 0.7
+// -------------------------------------------
+// SIDE 1 - SELL
+// -------------------------------------------
+// S = 1 - sqrt( 0.8*0.7 ) = 0.2516685226 --> 25%, valid cause is a number between 0 and 1
+// -------------------------------------------
+// SIDE 2 - SELL
+// -------------------------------------------
+// e.g
+// S = 1 + sqrt( 0.8*0.7 ) = 1.7483314774 --> Discarded, out of the range [0, 1]
+
+// *** BUY ***
+// ------------------
+// Pab = abOut/abIn
+// Pba = baOut/baIn[abIn]
+// S = 1 - sqrt( Pab/Pba )
+// OR
+// S = 1 + sqrt( Pab/Pba )
+// -------------------------------------------
+// EXAMPLE [BUY]
+// -------------------------------------------
+// 2970B > 122A // 122A >> 2400B
+// --> 2970/122 = 24.34
+// --> 2400/122 = 19.67
+// -------------------------------------------
+// SIDE 1 - BUY
+// -------------------------------------------
+// S = 1 - sqrt( 24.34/19.67 ) = 0.2516685226 --> -0.109 --> -10.9%, valid cause is a number between 0 and 1
```

Comment:
`-0.109` is not between `0` and `1` :thinking: 
Unless you mean the modulus?

Reply:
The 0...1 thingy is something I saw experimentally, but is likely not the way to choose the right solution

not sure, we should do more examples.

Reply:
Actually, thinking a bit more on this, I tried to justify the logic to pick the sign of the formula.

After a lot of thinking, I arrived to a conclusion, although this doesn't mean my conclusion is correct  😂

TLDR: We should always take the `MINUS` sign.

If you want the full explanation why I claim this:

![IMG_3551](https://user-images.githubusercontent.com/2352112/144136543-aa232018-f8a6-4059-bb1f-0eb9719c8c77.jpg)

![IMG_3552](https://user-images.githubusercontent.com/2352112/144136570-04488a7f-810a-4924-b4df-87d03e7cf391.jpg)